model/dns_zone/dns_zone.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require_relative "../../model"
+
+class DnsZone < Sequel::Model
+  many_to_one :project
+  many_to_many :dns_servers
+  one_to_many :records, class: :DnsRecord
+  one_to_many :active_billing_records, class: :BillingRecord, key: :resource_id, conditions: {Sequel.function(:upper, :span) => nil}
+
+  include ResourceMethods
+  include Authorization::HyperTagMethods
+  include Authorization::TaggableMethods
+
+  def hyper_tag_name(project)
+    "project/#{project.ubid}/location/#{location}/dns_zone/#{ubid}"
```

Comment:
We aim to that resources have the same tag with their URLs at both the API and UI. Users can easily copy the path and paste it into the access policies. While ﻿_ is commonly used in URLs, it is not fully compatible with all URL RFCs. https://stackoverflow.com/questions/7930751/regexp-for-subdomain/7933253#7933253

I checked the codebase; we have `private-subnet` and `minio_cluster`. I believe we decided to use the same naming convention for all our resource types. I prefer using `-` as a separator instead of `_`.

Reply:
Also we allow `-` in the name of resources, not `_`. I don't like to mixed them in the URLs

Reply:
@furkansahin We may need a minor update to `minio_cluster`.

Reply:
I'll update it